src/python/zquantum/core/distance_measures.py

```diff
@@ -0,0 +1,108 @@
+            x_i (np.array): Samples A (integers).
+            y_j (np.array): Samples B (integers).
+            sigma (float): The bandwidth of the gaussian kernel.
+
+        Returns:
+            np.ndarray: The gaussian kernel matrix.
+    """
+    exponent = np.abs(x_i[:, None] - y_j[None, :]) ** 2
+    try:
+        gamma = 1.0 / (2 * sigma)
+    except ZeroDivisionError as error:
+        print("Handling run-time error:", error)
+        raise
+    K = np.exp(-gamma * exponent)
+    return K
+
+
+def compute_multi_rbf_kernel(x_i, y_j, sigmas):
+    """ Compute the multi-gaussian (RBF) kernel matrix K, with K_ij = 1/c Sum_n[exp(-gamma_n |x_i - y_j|^2)] and gamma = 1/(2*sigma).
+
+        Args:
+            x_i (np.array): Samples A (integers).
+            y_j (np.array): Samples B (integers).
+            sigmas (np.array): The list of bandwidths of the multi-gaussian kernel.
+
+        Returns:
+            np.ndarray: The gaussian kernel matrix.
+    """
+    exponent = np.abs(x_i[:, None] - y_j[None, :]) ** 2
+    K = 0.0
+    for sigma in sigmas:
+        try:
+            gamma = 1.0 / (2 * sigma)
+        except ZeroDivisionError as error:
+            print("Handling run-time error:", error)
+            raise
+        K = K + np.exp(-gamma * exponent)
+    return K / len(sigmas)
+
+
+def compute_mmd(
+    target_distribution, measured_distribution, distance_measure_parameters,
+):
+    """ Compute the squared Maximum Mean Discrepancy (MMD) distance measure between between a target bitstring distribution
+    and a measured bitstring distribution.
+    Reference: arXiv.1804.04168.
+
+        Args:
+            target_distribution (BitstringDistribution): The target bitstring probability distribution.
+            measured_distribution (BitstringDistribution): The measured bitstring probability distribution.
+            distance_measure_parameters (dict): dictionary containing the relevant parameters for the maximum mean discrepancy, i.e.:
+                                                - sigma: the bandwidth parameter used to compute the gaussian kernel.
+        Returns:
+            float: The value of the maximum mean discrepancy.
+    """
+
+    sigma = distance_measure_parameters.get("sigma", 1.0)
+    target_keys = target_distribution.distribution_dict.keys()
+    measured_keys = measured_distribution.distribution_dict.keys()
+    all_keys = set(target_keys).union(measured_keys)
+
+    target_values = []
+    measured_values = []
+    for bitstring in all_keys:
+        # Add 0 to the values list whenever a bistrings isn't found among the keys.
+        target_values.append(target_distribution.distribution_dict.get(bitstring, 0))
+        measured_values.append(
+            measured_distribution.distribution_dict.get(bitstring, 0)
+        )
+
+    basis = np.asarray([int(item, 2) for item in all_keys])  # bitstring to int
+    if not hasattr(sigma, "__len__"):
+        K = compute_rbf_kernel(basis, basis, sigma)
```

Comment:
Can we rename it from `K` to something more meaningful?
I understand why it is called `K` in the other functions and why don't want to change it (though still think it's a bad variable name), but in our code the name should be more descriptive.

Reply:
Changed it to `kernel_matrix` everywhere (see [here](https://github.com/zapatacomputing/z-quantum-core/pull/63/commits/67915991fc62d165cf76ffa8b80fd848fc9f803d)). I don't think it's a problem to change it since we already made some modifications to the original code!